### PR TITLE
[Skeleton Tool] Creating new Skeleton layer without a bone

### PR DESCRIPTION
**Pull request by Aa20475**
Issue #1535  

This is the current state. I'm sure I was  unable to do this much before. I just need to add a grouper here. 

I also closed some edge cases here.

### CODE REVIEW

**Comment by morevnaproject** [comment]
> I just need to add a grouper here.

For PassiveGrouper: check example at `widget_timetrack.cpp`:

screenshot_001

Line (1) creates PassiveGrouper and all subsequent actions  from "for" loop below (2) will get added to that group automatically.

**Comment by morevnaproject** [comment]
I have tested this PR. It behaves weird. Sometimes it is works, but sometimes it is crashing for unexpected reason.
For same actions it sometimes crashes and sometimes it isn't.

Video - https://youtu.be/rCSzcRrvs9M

In any case, that's good to see progress on this issue. ^__^

**Comment by Aa20475** [comment]
The crash is because the `active_bone_` is not being initialised. I'll fix that!

**Comment by Aa20475** [comment]
Also, the conditions in `set_param` are a bit wrong in  `valuenodesetactivebone`. That's the major issue :sweat_smile:

**Comment by Aa20475** [comment]
Hey! I think it works fine now? Can you please test?

File: synfig-studio/src/gui/states/state_bone.cpp [hunk 1]
```diff
@@ -619,11 +619,7 @@ StateBone_Context::event_mouse_release_handler(const Smach::event& x)
 						list_node=ValueNode_StaticList::Handle::cast_dynamic(list_desc.get_value_node());
 						ValueDesc value_desc= ValueDesc(list_node,active_bone,list_desc);
 						setActiveBone->set_param("active_bone_node",value_desc.get_value_node());
-						if(get_work_area()->get_active_bone_value_node()){
-							setActiveBone->set_param("prev_active_bone_node",get_work_area()->get_active_bone_value_node());
-						}else{
-							setActiveBone->set_param("prev_active_bone_node",ValueNode::Handle::cast_dynamic(ValueNode_Bone::create(Bone())));
-						}
+						setActiveBone->set_param("prev_active_bone_node",get_work_area()->get_active_bone_value_node());
```
Comment (rodolforg) [inline synfig-studio/src/gui/states/state_bone.cpp:622]: Shouldn't action `ValueNodeSetActiveBone` set was the previous one automatically?

File: synfig-studio/src/gui/states/state_bone.cpp [hunk 2]
```diff
@@ -785,36 +800,64 @@ StateBone_Context::event_mouse_release_handler(const Smach::event& x)
...
+							setActiveBone->set_param("prev_active_bone_node",get_work_area()->get_active_bone_value_node());
+
+							if (setActiveBone->is_ready()){
+								try{
+									get_canvas_interface()->get_instance()->perform_action(setActiveBone);
 								} catch (...) {
 									info("Error performing action");
 								}
 							}
+							// Use Passive grouper
```
Comment (rodolforg) [inline synfig-studio/src/gui/states/state_bone.cpp:860]: ?

File: synfig-studio/src/gui/states/state_bone.cpp (hunk 1 above)
Comment (Aa20475) [reply synfig-studio/src/gui/states/state_bone.cpp:622]: Actually, as I was unable to access anything related to rendering i.e, `WorkArea`, `CanvasView` from actions, I decided that the user will provide it. :sweat_smile:

File: synfig-studio/src/gui/states/state_bone.cpp (hunk 2 above)
Comment (Aa20475) [reply synfig-studio/src/gui/states/state_bone.cpp:860]: Will remove my bad

**Comment by morevnaproject** [comment]
Hi! I am sorry to say that, but it is still crashing - https://youtu.be/RL4_2KenYjI

**Comment by morevnaproject** [comment]
It works good now. Thank you!

**Pull request merged**